bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/State.java

```diff
@@ -184,4 +189,62 @@ public Effect getEffect() {
     public boolean isReachable() {
         return reachable;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((alert == null) ? 0 : alert.hashCode());
+        result = prime * result + bri;
+        result = prime * result + ((colormode == null) ? 0 : colormode.hashCode());
+        result = prime * result + ct;
+        result = prime * result + ((effect == null) ? 0 : effect.hashCode());
+        result = prime * result + hue;
+        result = prime * result + (on ? 1231 : 1237);
+        result = prime * result + (reachable ? 1231 : 1237);
+        result = prime * result + sat;
+        result = prime * result + Arrays.hashCode(xy);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
One last thing, can you run this class through the formatter. It looks like the generated single-line if statements don't have brackets.

Reply:
I have used the formatter settings in Visual Studio Code as described on openhab documentation. It doesn't add brackets automatically.

Reply:
I added the brackets manually, now.

Reply:
It doesn't? Strange, that means that the VSC formatter settings is different from the Eclipse one. The Eclipse one is the standard so looks like the VSC formatter needs to be changed.